HW3.py

```diff
@@ -0,0 +1,116 @@
+        except KeyError:
+            result[char] = 1  # If char not in result dict yet, start counting it
+            sumFreqChars +=1
+    for char in result:
+        freq[char]=round(result[char]*100/sumFreqChars,2)
+
+    result = sorted(freq.items(), key=operator.itemgetter(1), reverse=True)
+    return result
+
+def newGame():
+    answer = input(ONE_MORE_TRY).upper()
+    if answer.startswith(YES):
+        start_game()
+    else:
+        print(BYE)
+        sys.exit()
+
+def start_game():
+    if len(sys.argv) < 2:
+        with open(PATH_TO_DATABASE) as f:
+            print(WELCOME)
+            given_word_length = int(input(THINK_A_WORD))
+            search_expr = DOT * given_word_length
+            word_subset = searchUniqueWords(searchWords(f),given_word_length)  # Select words with given length
+            tried_chars = []
+            GameOn = True
+            while GameOn:
+                char_counts = getCharFrequency(word_subset)
+                for x in range(len(char_counts)):
+                    char_counts = getCharFrequency(word_subset)
+                    mostFreqChar = char_counts[x][0] # Get most frequent char
+                    if mostFreqChar in tried_chars:
+                        continue  # Try next char if such a char already played
+                    else:
+                        tried_chars.append(mostFreqChar)
+                    occurencesList = input(POSITIONS_OF_AVAILABLE_LETTERS.format(mostFreqChar)).split(COMMA)
+                    if occurencesList != [EMPTY_STRING]:  # if some positions entered, form regexp for further word's filtering
+                        search_expr = list(search_expr)  # convert regexp from string to list for modification
+                        for pos in occurencesList:
+                            search_expr[int(pos)] = mostFreqChar
+                        search_expr = EMPTY_STRING.join(search_expr)  # convert regexp back from list to string
+                        r = re.compile(search_expr)
+                        word_subset = list(filter(r.match, word_subset))
+                        if len(word_subset) == 1:
+                            print(GUESSED_WORD.format(word_subset[0]))
+                            newGame()
+                        elif len(word_subset) == 0:
+                            print(CANT_GUESS)
+                            newGame()
+                        else:
+                            break  # There're still words to guess, retry with new char frequencies on new wordSubset
+                    else:
+                        if x == len(char_counts):
+                            print(NO_TRIES_LEFT)
+                            sys.exit()
+                        continue  # if nothing entered, than no char the word includes, so repeat with next most popular char
+    else:
+        with open(sys.argv[1], 'r') as f:
+            searchWords(f)                
+
+if __name__ == '__main__':
+    start_game()
```

Comment:
It'd be better to check if your code meets the PEP8 using pep8 module or [pep8online](http://pep8online.com/checkresult)

Happy coding :)